src/ansys/fluent/core/workflow.py

```diff
@@ -1348,33 +1430,24 @@ def load_state(self, list_of_roots: list):
         """Load the state of the workflow."""
         self._workflow.LoadState(ListOfRoots=list_of_roots)
 
-    def _populate_help_string_command_id_map(self):
-        if not self._help_string_command_id_map:
-            for command in dir(self._command_source):
-                if command in ["SwitchToSolution", "set_state"]:
-                    continue
-                command_obj = getattr(self._command_source, command)
-                if isinstance(command_obj, PyCommand):
-                    command_obj_instance = command_obj.create_instance()
-                    help_str = command_obj_instance.get_attr("helpString")
-                    if help_str in self.child_task_python_names():
-                        self._help_string_command_id_map[help_str] = command
-                        self._help_string_display_text_map[help_str] = (
-                            command_obj_instance.get_attr("displayText")
-                        )
-                    del command_obj_instance
-
-    def get_possible_tasks(self):
-        """Get the list of possible names of commands that can be inserted as tasks."""
-        self._populate_help_string_command_id_map()
-        return list(self._help_string_command_id_map)
-
-    def insert_new_task(self, command_name: str):
+    def get_insertable_tasks(self):
+        """Get the list of possible command names that can be inserted as tasks."""
+        return [
+            item
+            for item in self._help_string_command_id_map.keys()
+            if item not in self._repeated_task_help_string_display_text_map.keys()
+        ]
+
+    def get_available_task_names(self):
+        """Get the list of the Python names for the available tasks."""
+        return [child.python_name() for child in self.ordered_children()]
+
+    def insert_new_task(self, task: str):
```

Comment:
Please also update the docstring to omit mention of command. We can also remove the superfluous "passed as an argument".

Reply:
Done, Thank you. Removing mention of 'command' from BaseTask will be done as part of the next PR.